src/auth/auth.go

```diff
@@ -41,18 +41,28 @@ const (
 	// can be created for this user (during auth activation) and they cannot
 	// be removed from the set of cluster super-admins.
 	RootUser = "pach:root"
-)
 
-// ParseScope parses the string 's' to a scope (for example, parsing a command-
-// line argument.
-func ParseScope(s string) (Scope, error) {
-	for name, value := range Scope_value {
-		if strings.EqualFold(s, name) {
-			return Scope(value), nil
-		}
-	}
-	return Scope_NONE, errors.Errorf("unrecognized scope: %s", s)
-}
+	// PpsUser is a special, unrevokable cluster administrator account used by PPS
+	// to create pipeline tokens, close commits, and do other necessary PPS work.
+	// It's not possible to authenticate as ppsUser (pps reads the auth token for
+	// this user directly from etcd).
+	PpsUser = `pach:pps`
+
+	// ClusterAdminRole is the role for cluster admins, who have full access to all APIs
+	ClusterAdminRole = "clusterAdmin"
```

Comment:
No preference on my part, but just curious: why define roles in Go as strings, but permissions in the proto as enums? Is the thinking that roles may need more flexibility or multiple parts later on?

Reply:
Yeah, my thinking here is that user-defined roles are an inevitable next step. Having roles be strings we validate in the server paves the way for that. Permisisons are an enum because they're tightly coupled to the actual code.